spec/graphql/mutations/repository/clone_repository_mutation_spec.rb

```diff
@@ -59,39 +59,75 @@
 
   subject { result }
 
-  context 'remote is valid' do
+  context 'with a valid remote' do
     before do
       # Stub the Bringit::Wrapper
       allow(Bringit::Wrapper).
         to receive(:valid_remote?).
         and_return(true)
+
+      allow(::Repository).
+        to receive(:create).
+        with('name' => repository_data['name'],
+             'description' => repository_data['description'],
+             'owner' => user,
+             'content_type' => repository_data['contentType'],
+             'public_access' => repository_data['visibility'] == 'public',
+             'remote_address' => variables['remoteAddress'],
+             'remote_type' => variables['remoteType']).
+        and_call_original
     end
     it 'enqueues a clone-repository job' do
       expect(RepositoryCloningJob).
         to have_been_enqueued.
         with(subject['data']['cloneRepository']['id'])
     end
+
+    it 'creates the repository' do
+      expect(subject).not_to be(nil)
+    end
+
+    it 'creates the url mappings' do
+      allow(UrlMapping).to receive(:create).and_call_original
+      subject
+      url_mappings.each do |url_mapping|
+        expect(UrlMapping).
+          to have_received(:create).
+          with(repository_id:
+                 Repository.first(slug: repository_blueprint.to_param).id,
+               source: url_mapping['source'],
+               target: url_mapping['target'])
+      end
+    end
   end
 
-  context 'remote is invalid' do
+  context 'with an invalid remote' do
     before do
       # Stub the Bringit::Wrapper
       allow(Bringit::Wrapper).
         to receive(:valid_remote?).
         and_return(false)
+
+      allow(UrlMapping).
+        to receive(:create)
     end
 
-    it 'enqueues a clone-repository job' do
+    it 'does not enqueue a clone-repository job' do
       expect(RepositoryCloningJob).
         not_to have_been_enqueued
     end
 
-    it 'error has been added to the graphql context' do
+    it 'adds a GraphQL error' do
       expect(subject.to_h).
         to match('data' => {'cloneRepository' => nil},
                  'errors' =>
                    [include('message' =>
                      /remote_address: ".*" is not a git or svn repository/)])
     end
+
+    it 'repository was not created' do
+      expect(UrlMapping).
+        not_to have_received(:create)
```

Comment:
This is a way to check if the repository has been created. A more intuitive way would be:
```ruby
expect(Repository.first(slug: repository_blueprint.to_param)).to be(nil)
```
... because you check the existence of the *repository* instead of the url mapping.